rbi/stdlib/rubygems.rbi

```diff
@@ -2634,27 +2634,36 @@ class Gem::Version < Object
 
   sig do
     params(
-      version: T.any(String, Gem::Version),
-    )
-    .void
+      version: T.any(
+        String,
+        Integer,
+        Float, # Technically, any `to_s`able object would work.
+        Gem::Version, # Will return the same version object
+        NilClass # Depracated, and will issue a warning
+      )
+    ).void
   end
   def initialize(version); end
 
   def _segments(); end
 
   def _split_segments(); end
```

Comment:
I left some of these out because they're intended to be psuedo-private, and have a complicated shape like `[T:Array[Integer], T::Array[T.any(Integer, String)]]`